dao/src/main/java/org/wso2/testgrid/dao/uow/TestPlanUOW.java

```diff
@@ -172,6 +172,16 @@ public TestPlan getLastFailure(TestPlan testPlan) {
         return testPlanRepository.getTestPlanHistory(testPlan);
     }
 
+    /**
+     * Returns a list of testplans that needed to be deleted
+     * @param productName job name
+     * @param count number of datasource that need to be saved in each infra combination
+     * @return list of testplans
+     */
+    public List<String> getDeletingDataSources(String productName, int count) {
```

Comment:
I think the unit of work is delete datasources. So, I would expect that there is a method named like deleteDatasourcesByAge.